src/ip_data.py

```diff
@@ -232,7 +300,7 @@ def subnet_target(self) -> Iterator[list[Any]]:
 
 def gen_db(dbidx: int) -> None:
     with MockDB(dbidx) as mdb:
-        mdb.init_tables().write_tables().log_answer()
+        mdb.init_tables().write_tables(SqliteWriter).log_answer()
```

Comment:
alright, so since you've decided todo dependency injection, maybe it should be `mdb.with_writer(SqliteWrite).init_tables().write_tables()` instead